combat-trainer.lic

```diff
@@ -1602,7 +1600,9 @@ class SpellProcess
                    .select { |spell| spell['slivers'] ? DRSpells.slivers : true }
                    .select { |spell| spell['starlight_threshold'] ? enough_starlight?(game_state, spell) : true }
     data = if @offensive_spell_cycle.empty?
-             ready_spells.min_by { |spell| DRSkill.getxp(spell['skill']) }
+             skills = ready_spells.map { |spell| spell['skill'] }
+             skill = game_state.sort_by_rate_then_rank(skills).first
+             ready_spells.find { |spell| spell['skill'] == skill }
```

Comment:
Original logic is being updated to sort by rank, too. As well as to make that decision consistent throughout combat-trainer by using game_state.sort_by_rate_then_rank. If the decision algoritm should change in the future, there's now one spot to do it.